tests/Unit/DSCResources/DSC_PostgreSqlScript.tests.ps1

```diff
@@ -0,0 +1,196 @@
+[System.Diagnostics.CodeAnalysis.SuppressMessageAttribute('PSAvoidUsingConvertToSecureStringWithPlainText', '')]
+param()
+
+$script:dscModuleName   = 'PostgreSqlDsc'
+$script:dscResourceName = 'DSC_PostgreSqlScript'
+
+function Invoke-TestSetup
+{
+    try
+    {
+        Import-Module -Name DscResource.Test -Force -ErrorAction 'Stop'
+    }
+    catch [System.IO.FileNotFoundException]
+    {
+        throw 'DscResource.Test module dependency not found. Please run ".\build.ps1 -Tasks build" first.'
+    }
+
+    $script:testEnvironment = Initialize-TestEnvironment `
+        -DSCModuleName $script:dscModuleName `
+        -DSCResourceName $script:dscResourceName `
+        -ResourceType 'Mof' `
+        -TestType 'Unit'
+}
+
+function Invoke-TestCleanup
+{
+    Restore-TestEnvironment -TestEnvironment $script:testEnvironment
+}
+
+Invoke-TestSetup
+
+try
+{
+    InModuleScope $script:dscResourceName {
+        $moduleResourceName = 'PostgreSqlDsc - DSC_PostgreSqlScript'
+
+        $superAccountCred = New-Object `
+        -TypeName System.Management.Automation.PSCredential `
+        -ArgumentList 'postgresqlAdmin', (ConvertTo-SecureString 'DummyPassword' -AsPlainText -Force)
+
+        $scriptParams = @{
+            DatabaseName     = 'testdb'
+            SetFilePath      = "C:\test.sql"
```

Comment:
I would change each of these to have Set, Get, and Test in the name of the .sql file to ensure that the correct parameter is called in the various areas.